megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -1330,18 +1330,21 @@ private void updateWhenSelected() {
     }
 
     public void boardNew() {
-        RandomMapDialog rmd = new RandomMapDialog(frame, this, null, mapSettings);
-        rmd.setVisible(true);
-        board = BoardUtilities.generateRandom(mapSettings);
-        game.setBoard(board);
-        curfile = null;
-        frame.setTitle(Messages.getString("BoardEditor.title")); //$NON-NLS-1$
-        menuBar.setBoard(true);
-        bvc.doLayout();
+    	RandomMapDialog rmd = new RandomMapDialog(frame, this, null, mapSettings);
+    	boolean userCancel = rmd.activateDialog();
+    	if (!userCancel) {
```

Comment:
```suggestion
    	if (!rmd.activateDialog()) {
```

Reply:
I'd rather waste that one line for improved readability.